### PR TITLE
Anomaly detection module updates

### PR DESCRIPTION
**Pull request by jacobmontiel**
Changes proposed in this pull request:

- Add `AnomalySineGenerator`.
- Fix bug in `HalfSpaceTrees` raised when predictions are requested and the model is empty.
- Update usage example in `HalfSpaceTrees` docstring to use the new `AnomalySineGenerator`.
- Add new tests for `AnomalySineGenerator`.
- Parameter `y` in `partial_fit` is no longer a requirement as it is not actually used. We keep it in the signature only for compatibility.

---

#### Implementation
- [x] Implementation is correct (it performs its intended function).
- [x] Code is consistent with the framework.
- [x] Code is properly documented.
- [x] PR description covers ALL the changes performed.
- [x] Files changed (update, add, delete) are in the PR's scope (no extra files are included).

#### Tests
- [x] New functionality is tested.
- [x] Tests are created for the new functionality or existing tests are updated accordingly.
- [x] ALL tests pass with no errors.
- [x] CI/CD pipelines run with no errors.
- [x] Test Coverage is maintained (coverage may drop by no more than 0.2%).

### CODE REVIEW

File: tests/data/test_anomaly_sine_generator.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    X_expected = data['X']
+    y_expected = data['y']
+
+    X, y = stream.next_sample()
+    assert np.alltrue(np.isclose(X[0], X_expected[0]))
+    assert np.alltrue(np.isclose(y[0], y_expected[0]))
+
+    X, y = stream.last_sample()
+    assert np.alltrue(np.isclose(X[0], X_expected[0]))
+    assert np.alltrue(np.isclose(y[0], y_expected[0]))
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply tests/data/test_anomaly_sine_generator.py:53]: `last_sample` means last **seen** sample in the context of the stream. It is just to recall what was seen in the last `next_sample` operation.

File: src/skmultiflow/data/anomaly_sine_generator.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            Return a tuple with the features X and the target y for
+            the batch_size samples that are requested.
+
+        """
+        self.sample_idx += batch_size
+        try:
+            self.current_sample_x = self.X[self.sample_idx - batch_size:self.sample_idx, :]
+            self.current_sample_y = self.y[self.sample_idx - batch_size:self.sample_idx]
+            if self.n_targets < 2:
+                self.current_sample_y = self.current_sample_y.flatten()
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply src/skmultiflow/data/anomaly_sine_generator.py]: Indeed, removed in f957d3ec3b1b10d9d2e2b9439d2c1febabf93320

File: src/skmultiflow/data/anomaly_sine_generator.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+        except IndexError:
+            self.current_sample_x = None
+            self.current_sample_y = None
+
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply src/skmultiflow/data/anomaly_sine_generator.py]: Good catch, this has been updated in f957d3ec3b1b10d9d2e2b9439d2c1febabf93320

File: tests/data/test_anomaly_sine_generator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (smastelini) [reply tests/data/test_anomaly_sine_generator.py:53]: My bad! It totally makes sense. I going to mark this one as solved. Thank you for clarifying :)

File: src/skmultiflow/data/anomaly_sine_generator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (smastelini) [reply src/skmultiflow/data/anomaly_sine_generator.py]: Thanks for addressing that!

File: src/skmultiflow/data/anomaly_sine_generator.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (smastelini) [reply src/skmultiflow/data/anomaly_sine_generator.py]: Nice workaround!

**Pull request merged**